pkg/ebpftracer/decoder/decoder.go

```diff
@@ -677,3 +677,24 @@ func Print16BytesSliceIP(in []byte) string {
 	ip := net.IP(in)
 	return ip.String()
 }
+
+var ErrUnknownFlowDirection = errors.New("unknown flow direction")
+
+func (decoder *Decoder) ReadFlowDirection() (types.FlowDirection, error) {
+	var rawDirection uint8
+
+	if err := decoder.DecodeUint8(&rawDirection); err != nil {
+		return types.FlowDirectionUnknown, err
+	}
+
+	direction := types.FlowDirection(rawDirection)
+
+	switch direction {
+	case types.FlowDirectionIncoming, types.FlowDirectionOutgoing:
+		return direction, nil
+	}
+
+	decoder.log.Errorf("unknown flow direction: %d", rawDirection)
```

Comment:
Logging in hot path is not ideal. We have metric for decoder errors. Would be better to then extend that metric with some reason label.

Reply:
I thought about this as well. The reason I still went with it, is that we anyway return an error and this should be caught in development.